hazelcast/src/main/java/com/hazelcast/sql/impl/expression/string/LikeFunction.java

```diff
@@ -54,18 +57,26 @@
     /** Special characters which require escaping in Java. */
     private static final String ESCAPE_CHARACTERS_JAVA = "[]()|^+*?{}$\\.";
 
+    private boolean negated;
     private transient State state;
 
     public LikeFunction() {
```

Comment:
Question: do we use java reflections to instantiate so that we have to have empty `public` constructor?

Reply:
The public constructor is required by our serialization engine, see `SqlDataSerializerHook`.